eslint-plugin-kuc-v1/validator-in-update.js

```diff
@@ -0,0 +1,45 @@
+module.exports = {
+  create: function(context) {
+    return {
+      ClassDeclaration: function(node) {
+        const superClass = node.superClass.name;
+        if (superClass !== "KucBase") return;
+
+        const nodeBody = node.body;
+        if (nodeBody.type !== "ClassBody") return;
+
+        const sourceCode = context.getSourceCode();
+        const pattern = "validate[a-zA-z]+\\(";
+        const regex = new RegExp(pattern);
+        if (!regex.test(sourceCode.getText())) return;
+
+        nodeBody.body.forEach(body => {
+          if (
+            body.type === "MethodDefinition" &&
+            body.key.type === "Identifier" &&
+            !body.computed
+          ) {
+            const functionSourceCode = sourceCode.getText(body.value);
+            if (
+              (body.key.name === "update" && !regex.test(functionSourceCode)) ||
+              (body.kind !== "constructor" &&
+                body.key.name !== "update" &&
+                regex.test(functionSourceCode))
+            ) {
+              context.report({
+                node: body,
+                message: `Please call validator function inside update() lifecycle
```

Comment:
I added the note on `super.update()`.
Also on the rule pattern, is it possible to check that there is super.update method when implementing validator on update lifecycle?

```suggestion
                message: `Please call validator function inside update() lifecycle and write super.update(changedProperties) using PropertyValues to map the changed properties with old values.
```